docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+
+**Example:**
+
+```csharp
+public interface IUserInterface
+{
+    public int InterfaceProperty { get; set; }
+}
+
+public interface IUserInterface2
+{
+    public float InterfacePropertyOnlyGetter { get; }
+}
+```
+
+```csharp
+[AutoImplement(nameof(IUserInterface), nameof(IUserInterface2))]
+public partial class UserClass
+{
+    public string UserProp { get; set; }
+}
+```
+
+```csharp
+#nullable enable
+[Generator]
+public class AutoImplementGenerator : IIncrementalGenerator
+{
+    private const string NameOfString = "nameof(";
+    private const string AutoImplementAttributeNameSpace = "AttributeGenerator";
+    private const string AutoImplementAttributeName = "AutoImplement";
+    private const string AutoImplementAttributeClassName = $"{AutoImplementAttributeName}Attribute";
+    private const string FullyQualifiedMetadataName = $"{AutoImplementAttributeNameSpace}.{AutoImplementAttributeClassName}";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(ctx =>
+        {
+            //Generate the AutoImplement Attribute
+            const string autoImplementAttributeDeclarationCode = $$"""
+using System;
+namespace {{AutoImplementAttributeNameSpace}};
+
+[AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+sealed class {{AutoImplementAttributeClassName}} : Attribute
+{
+    public string[] InterfacesNames { get; }
+    public {{AutoImplementAttributeClassName}}(params string[] interfacesNames)
+    {
+        InterfacesNames = interfacesNames;
+    }
+}
+""";
+            ctx.AddSource($"{AutoImplementAttributeClassName}.g.cs", autoImplementAttributeDeclarationCode);
+        });
+
+        IncrementalValuesProvider<Model> provider = context.SyntaxProvider.ForAttributeWithMetadataName(
+            fullyQualifiedMetadataName: FullyQualifiedMetadataName,
+            predicate: static (node, cancellationToken_) => node is ClassDeclarationSyntax,
+            transform: static (ctx, cancellationToken) =>
+            {
+                ClassDeclarationSyntax classDeclarationSyntax = (ClassDeclarationSyntax)ctx.TargetNode;
+                
+                string className = classDeclarationSyntax.Identifier.ValueText;
+                string classNameSpace = GetClassNameSpace(classDeclarationSyntax.Parent);
+                InterfaceModel[] interfaces = GetInterfaceModels(ctx.SemanticModel.Compilation, classDeclarationSyntax);
+
+                return new Model(
+                    className,
+                    classNameSpace,
+                    interfaces
+                    );
+            }).Where(m => m is not null);
+
+        context.RegisterSourceOutput(provider.Collect(), Execute);
```

Comment:
Don't collect here; just pass in `provider` directly, and remove the outer `foreach` from `Execute`. I'd also rename `Execute` to be something more descriptive, rather than letting it be a relic of `ISourceGenerator`.